spec/unit/file_system_spec.rb

```diff
@@ -25,29 +25,44 @@ def with_file_content(content)
   end
 
   context "#open" do
-    it "uses a default mode of 100644, the same as File.open, when specifying a nil mode" do
+    it "uses the same default mode as File.open, when specifying a nil mode (umask used on non-Windows)" do
       file = tmpfile('file_to_update')
       expect(Puppet::FileSystem.exist?(file)).to be_falsey
 
       Puppet::FileSystem.open(file, nil, 'a') { |fh| fh.write('') }
 
-      # default mode of 0644 is applied
-      expect(File.stat(file).mode.to_s(8)).to eq('100644')
+      expected_perms = Puppet::Util::Platform.windows? ?
+        # default Windows mode based on temp file storage
+        '2000700' :
+        # default mode is applied. 100 == 'regular file'
+        '100' + (666 - File.umask.to_s(8).to_i).to_s
```

Comment:
Yes! Thanks @joshcooper! That's exactly what I wanted to do when I put that in the first time. PR inbound.